### PR TITLE
Updated MF app and tests files (WIP)

### PR DESCRIPTION
**Pull request by shivampdesai**
Do not merge

### CODE REVIEW

**Comment by neoblizz** [comment]
Quick note before review, make your commit comments, ones in: `git commit -m "comments"`, more descriptive. It helps us figure out what changed in that specific commit. It will be useful for you too in future when you go back and look at your code. 👍

File: gunrock/app/mf/mf_test.cuh [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+ * @param[out]  preds         Computed predecessors for each vertex
+ * @param[in]   src           The source vertex
+ * @param[in]   quiet         Whether to print out anything to stdout
+ * @param[in]   mark_preds    Whether to compute predecessor info
+ * \return      double        Time taken for the SSSP
+ */
+template <
+    typename GraphT,
+    typename ValueT = typename GraphT::ValueT>
+double CPU_Reference(
```
Comment (neoblizz) [inline gunrock/app/mf/mf_test.cuh]: I think this is the next step for you, understanding how to add your CPU reference code (that you wrote a while ago). Incorporating this within the gunrock layer will allow us to use it for testing later and also help us write what goes inside of the advance and filter operator.

File: gunrock/app/mf/mf_test.cuh [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    //    [](const PairT &a, const PairT &b) -> bool
+    //    {
+    //        return a.first < b.first;
+    //    });
+    // for (VertexT v = 0; v < graph.nodes; ++v)
+    //    distances[v] = sort_dist[v].second;
+    delete[] sort_dist; sort_dist = NULL;
+
+    return elapsed;
+#else
```
Comment (neoblizz) [inline gunrock/app/mf/mf_test.cuh]: You don't need boost reference at all for this, the above code starting from line#87 till line#161 is useless.

File: gunrock/app/mf/mf_test.cuh [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    //            distances[u] = u_distance;
+    //            pq.push(std::make_pair(u, u_distance));
+    //        }
+    //    }
+    // }
+    cpu_timer.Stop();
+    float elapsed = cpu_timer.ElapsedMillis();
+
+    return elapsed;
+#endif
```
Comment (neoblizz) [inline gunrock/app/mf/mf_test.cuh]: Remove this `#endif` if you end up removing the boost part, and copy paste/implement your cpu version here. Follow the example in sssp if you need a reference.
More or less it is taking your code and making sure it matches gunrock stuff. Don't worry about the cpu timers.

**Comment by neoblizz** [comment]
@agalup

**Comment by sgpyc** [comment]
Merge to let @agalup working on it

**Pull request merged**